### PR TITLE
Indent names in logger

### PR DESCRIPTION
**Pull request by vlkale**
# Summary of Change Made

This is a simple change for a better visual output of the kernel logger.

Each Kokkos region's label that is printed by the logger ought to be easily identified. A space is added in front each label identified by the variable 'name' for each Kokkos function having a tool callback in the code file debugging/kernel-logger/kp_kernel_logger.cpp. 

# Demonstration of Fix 

The following demonstrates this fix by showing a sample before / after ouput. The Kokkos Stream benchmark with 20 trials is run with clang++ on a  Mac OSX.

## Before Change

```
export KOKKOS_TOOLS_LIBS=${KokkosProjectHOME}/kokkos-tools/debugging/kernel-logger/kp_kernel_logger.so; ./stream.exe 
-------------------------------------------------------------
Kokkos STREAM Benchmark
-------------------------------------------------------------
KokkosP: Example Library Initialized (sequence is 0, version: 20211015)
Reports fastest timing per kernel
Creating Views...
Memory Sizes:
- Array Size:    100000000
- Per Array:           800.00 MB
- Total:              2400.00 MB
Benchmark kernels will be performed for 20 iterations.
-------------------------------------------------------------
KokkosP: Allocate<Host> name: a pointer: 0x12ae00040 size: 800000000
KokkosP: Executing fence on device 100663296 with unique execution identifier 0
KokkosP: Kokkos::Tools::invoke_kokkosp_callback: Kokkos Profile Tool Fence
KokkosP: Execution of fence 0 is completed.
KokkosP: Executing parallel-for kernel on device 100663297 with unique execution identifier 1
KokkosP: Kokkos::View::initialization [a] via memset
KokkosP: Executing fence on device 100663297 with unique execution identifier 2
KokkosP: HostSpace fence
KokkosP: Execution of fence 2 is completed.
KokkosP: Executing fence on device 100663296 with unique execution identifier 3
KokkosP: Kokkos::Tools::invoke_kokkosp_callback: Kokkos Profile Tool Fence
KokkosP: Execution of fence 3 is completed.
KokkosP: Execution of kernel 1 is completed.
KokkosP: Executing fence on device 100663297 with unique execution identifier 4
KokkosP: Kokkos::Impl::ViewValueFunctor: View init/destroy fence
KokkosP: Execution of fence 4 is completed.

```
## After Change

```
bash-5.1$ export KOKKOS_TOOLS_LIBS=${KokkosProjectHOME}/kokkos-tools/myBuild/debugging/kernel-logger/kp_kernel_logger.so; ./stream.exe
-------------------------------------------------------------
Kokkos STREAM Benchmark
-------------------------------------------------------------
KokkosP: Example Library Initialized (sequence is 0, version: 20211015)
Reports fastest timing per kernel
Creating Views...
Memory Sizes:
- Array Size:    100000000
- Per Array:           800.00 MB
- Total:              2400.00 MB
Benchmark kernels will be performed for 20 iterations.
-------------------------------------------------------------
KokkosP: Allocate<Host> name: a pointer: 0x112e00040 size: 800000000
KokkosP: Executing fence on device 100663296 with unique execution identifier 0
KokkosP:  Kokkos::Tools::invoke_kokkosp_callback: Kokkos Profile Tool Fence
KokkosP: Execution of fence 0 is completed.
KokkosP: Executing parallel-for kernel on device 100663297 with unique execution identifier 1
KokkosP:  Kokkos::View::initialization [a] via memset
KokkosP: Executing fence on device 100663297 with unique execution identifier 2
KokkosP:  HostSpace fence
KokkosP: Execution of fence 2 is completed.
KokkosP: Executing fence on device 100663296 with unique execution identifier 3
KokkosP:  Kokkos::Tools::invoke_kokkosp_callback: Kokkos Profile Tool Fence
KokkosP: Execution of fence 3 is completed.
KokkosP: Execution of kernel 1 is completed.
KokkosP: Executing fence on device 100663297 with unique execution identifier 4
KokkosP:  Kokkos::Impl::ViewValueFunctor: View init/destroy fence
KokkosP: Execution of fence 4 is completed.

```

It is expected to work on any platform; any inconsistencies with the above ouput on other platforms will be tracked and fixed accordingly. No change to performance or correctness is expected on any platform. 

This PR does not include the fix to filter out undesirable (tool-induced) fences from PR #152.

### CODE REVIEW

**Comment by dalg24** [comment]
Please post a before/after sample output

**Comment by vlkale** [comment]
> Please post a before/after sample output

@dalg24 See description of issue which has been edited to have this

File: profiling/memory-hwm-mpi/kp_hwm_mpi.cpp [hunk 1]
```diff
@@ -71,7 +71,7 @@ extern "C" void kokkosp_init_library(const int loadSeq,
   MPI_Comm_size(MPI_COMM_WORLD, &world_size);
 
   if (world_rank == 0) {
-    printf("KokkosP: Example Library Initialized (sequence is %d, version: %" PRIu64 ")\n", loadSeq, interfaceVer);
+    printf("KokkosP: High Water Mark Library Initialized (sequence is %d, version: %" PRIu64 ")\n", loadSeq, interfaceVer);
```
Comment (dalg24) [inline profiling/memory-hwm-mpi/kp_hwm_mpi.cpp:74]: This change does not belong to this PR

**Review by dalg24 — COMMENTED** [review]
Can you comment on why you decided to indent with a single space?

**Comment by crtrott** [comment]
I also think 2 or even 4 spaces might be better. And yeah the fix in the kp_hwm_mpi should be in a separate PR.

File: profiling/memory-hwm-mpi/kp_hwm_mpi.cpp (hunk 1 above)
Comment (crtrott) [reply profiling/memory-hwm-mpi/kp_hwm_mpi.cpp:74]: I think this is a targeting master vs develop thing maybe?

Comment (vlkale) [reply profiling/memory-hwm-mpi/kp_hwm_mpi.cpp:74]: @crtrott cc. @dalg24  Yes, I ought to have committed to develop branch (first), and not master. I forgot about that. I will be opening and linking here a new PR against the develop branch shortly.

**Comment by vlkale** [comment]
> Can you comment on why you decided to indent with a single space?

I thought that was enough for a user to  identify it. Looking at it closer, I agree it doesn't look all that apparent, especially given that there could be lengthy prints per line. Perhaps a command-line UI expert is needed to figure this out :) 

Also, I am not using a tab as the amount of space can be different on different machines. (The number of spaces in tab can actually be configurable for a machine, e.g., see https://www.ibm.com/docs/en/rational-clearcase/9.0.0?topic=cadmd-set-number-spaces-in-tab-character-unix-system-linux)

**Comment by crtrott** [comment]
Closing because we merged into develop
